e2e/test/scenarios/dashboard/reproductions/29304-scalar-first-render-size-change.cy.spec.js

```diff
@@ -0,0 +1,110 @@
+import { restore } from "e2e/support/helpers";
+
+import { SAMPLE_DATABASE } from "e2e/support/cypress_sample_database";
+
+const { ORDERS, ORDERS_ID } = SAMPLE_DATABASE;
+
+const SCALAR_QUESTION = {
+  name: "Scalar question",
+  query: {
+    "source-table": ORDERS_ID,
+    aggregation: [["count"]],
+  },
+  display: "scalar",
+};
+
+const SCALAR_QUESTION_CARD = { size_x: 4, size_y: 3, row: 0, col: 0 };
+
+const SMART_SCALAR_QUESTION = {
+  name: "Smart scalar question",
+  query: {
+    "source-table": ORDERS_ID,
+    aggregation: [["count"]],
+    breakout: [
+      [
+        "field",
+        ORDERS.CREATED_AT,
+        {
+          "base-type": "type/DateTime",
+          "temporal-unit": "month",
+        },
+      ],
+    ],
+  },
+  display: "smartscalar",
+};
+
+const SMART_SCALAR_QUESTION_CARD = SCALAR_QUESTION_CARD;
+
+describe("issue 29304", () => {
+  describe("display: scalar", () => {
+    beforeEach(() => {
+      restore();
+      cy.signInAsAdmin();
+      cy.intercept("api/dashboard/*/dashcard/*/card/*/query").as(
+        "getDashcardQuery",
+      );
+      cy.intercept("api/dashboard/*").as("getDashboard");
+      cy.clock();
+    });
+
+    it("should render scalar with correct size on the first render (metabase#29304)", () => {
+      cy.createDashboard().then(({ body: dashboard }) => {
+        cy.createQuestionAndAddToDashboard(
+          SCALAR_QUESTION,
+          dashboard.id,
+          SCALAR_QUESTION_CARD,
+        );
+
+        visitFullAppEmbeddingUrl({ url: `/dashboard/${dashboard.id}` });
+
+        cy.wait("@getDashboard");
+        cy.wait("@getDashcardQuery");
+        // The timeout necessary to make sure the dashcard is rendered.
+        const DASHCARD_FIRST_RENDER_TIMEOUT = 400;
+        cy.tick(DASHCARD_FIRST_RENDER_TIMEOUT);
```

Comment:
This `400` got me puzzled so I investigated a bit. It turns out it will work with `300`, but not `299`.

The value of this timeout must be greater or equal to [`WAIT_TIME` defined in `ExplicitSize`](https://github.com/metabase/metabase/blob/110b682/frontend/src/metabase/components/ExplicitSize.jsx#L11).

I suggest to import that `WAIT_TIME` here and reuse it, to clearly indicate the reason and intention.

Reply:
I don't fully understand this bit, to be honest. Since the new code uses `debounceLeading` the first render should not have to wait for the `WAIT_TIME` at all, I can understand if needs to wait for the `WAIT_TIME` if we use `debounce`.

Maybe I'll have to investigate and see why 300ms is still applied to `debounceLeading` mode.

Reply:
Fixed in 1765c3615028fb5cd2009a4e11b3cd21b397fa77.

Basically, it was exactly the `WAIT_TIME` that cause this delay. I've tested locally and both `debounce` and `debounceLeading` needs to wait exactly `WAIT_TIME` before it starts rendering in Cypress. I'm not sure why. But I've imported `WAIT_TIME` to use in the test to communicate it's `WAIT_TIME` that we're using in the test.